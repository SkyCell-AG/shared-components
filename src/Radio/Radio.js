import React, {
    useCallback,
} from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import {
    FormControlLabel,
    RadioGroup,
    Radio as MaterialRadio,
} from '@material-ui/core'

import Label from 'Label'

import useStyles from './Radio.styles'

const propTypes = {
    onChange: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    title: PropTypes.string,
    options: PropTypes.arrayOf(PropTypes.shape({
        value: PropTypes.string,
        label: PropTypes.string,
    })),
    name: PropTypes.string,
    value: PropTypes.string,
    compact: PropTypes.bool,
    small: PropTypes.bool,
    className: PropTypes.string,
    showInColumns: PropTypes.bool,
}

const defaultProps = {
    disabled: false,
    title: '',
    options: [],
    value: '',
    name: 'labelType',
    compact: false,
    small: false,
    className: '',
    showInColumns: false,
}

const Radio = (props) => {
    const {
        onChange,
        disabled,
        title,
        options,
        name,
        value,
        compact,
        small,
        className,
        showInColumns,
    } = props

    const classes = useStyles()

    const handleOnChange = useCallback((event) => {
        onChange(event.target.value, event)
    }, [onChange])

    return (
        <>
            <Label title={title} />
            <RadioGroup
                aria-label="labelType"
                name={name}
                value={value}
                className={clsx({
                    [classes.radio]: true,
                    [className]: true,
                    [classes.showInColumns]: showInColumns,
                })}
                onChange={handleOnChange}
                disabled={disabled}
                data-testid={`radio-${name}`}
            >
                {options.map((radio) => {
                    return (
                        <FormControlLabel
                            key={`radio-${radio.value}`}
                            value={radio.value}
                            className={clsx({
                                [classes.formControlStyle]: true,
                                [classes.radioSmall]: small,
                            })}
                            control={(
                                <MaterialRadio
                                    color="primary"
                                    classes={{
                                        root: compact && classes.radioRoot,
                                    }}
                                />
                            )}
                            data-value={radio.value}
                            label={radio.label}
                            disabled={disabled}
                        />
                    )
                })}
            </RadioGroup>
        </>
    )
}

Radio.propTypes = propTypes
Radio.defaultProps = defaultProps

export default Radio
