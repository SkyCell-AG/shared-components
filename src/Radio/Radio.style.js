import {
    makeStyles,
} from '@mui/styles'
import {
    useTheme,
} from '@mui/material/styles'

const useStyles = makeStyles(() => {
    const theme = useTheme()

    return {
        radio: {
            width: '100%',
            justifyContent: 'flex-start',
            display: 'flex',
            flexDirection: 'row',
            whiteSpace: 'nowrap',
            paddingLeft: '0px !important',
            paddingRight: '0px !important',
        },
        radioRoot: {
            padding: '1px 3px !important',
        },
        radioSmall: {
            fontSize: `${theme.typography.button.fontSize}px !important`,
            marginRight: theme.spacing(0),
            marginLeft: '0 !important',
        },
        formControlStyle: {
            marginLeft: theme.spacing(-0.625),
        },
        showInColumns: {
            flexDirection: 'column',
        },
        radioItemLabel: {},
        requiredMark: {
            position: 'absolute',
            left: -9,
            width: 6,
            height: 27,
            backgroundColor: theme.palette.primary.main,
            display: 'none',
            borderRadius: 3,
        },
        isRequired: {
            '& $requiredMark': {
                display: 'block',
            },
            backgroundColor: theme.palette.primary[30],
        },
    }
})

export default useStyles
