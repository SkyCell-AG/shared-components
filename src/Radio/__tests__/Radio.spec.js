/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import {
    shallow,
} from 'enzyme'
import {
    RadioGroup,
} from '@mui/material'

import Radio from '../Radio'

const options = [
    {
        value: 'option1',
        label: 'label1',
    },
    {
        value: 'option2',
        label: 'label2',
    },
]

describe('app/shared-components/Radio should match snapshot 1', () => {
    it('should match snapshot', () => {
        const props = {
            options,
            onChange: jest.fn(),
            title: 'Temperature Range',
            value: 'value',
            disabled: false,
        }
        const wrapper = shallow(
            <Radio
                {...props}
            />,
        )

        expect(wrapper).toMatchSnapshot()
    })
    it('should call onChange on radio group selected', () => {
        const onChangeMock = jest.fn()
        const wrapper = shallow(
            <Radio
                onChange={onChangeMock}
                value="option1"
                options={options}
                compact
                required
            />,
        )

        wrapper.find(RadioGroup).props().onChange({
            target: {
                value: 'option2',
            },
        })
        expect(onChangeMock).toBeCalledWith('option2', {
            target: {
                value: 'option2',
            },
        })
    })
})
