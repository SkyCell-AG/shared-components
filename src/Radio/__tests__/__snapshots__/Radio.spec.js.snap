// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/Radio should match snapshot 1 should match snapshot 1`] = `
<Fragment>
  <Label
    className=""
    hasError=""
    title="Temperature Range"
  />
  <div
    className=""
  >
    <div
      className="makeStyles-requiredMark-7"
    />
    <ForwardRef(RadioGroup)
      aria-label="labelType"
      className="makeStyles-radio-1 "
      data-testid="radio-labelType"
      disabled={false}
      name="labelType"
      onChange={[Function]}
      value="value"
    >
      <ForwardRef(FormControlLabel)
        className="makeStyles-formControlStyle-4"
        classes={
          Object {
            "label": "makeStyles-radioItemLabel-6",
          }
        }
        control={
          <ForwardRef(Radio)
            classes={
              Object {
                "root": false,
              }
            }
            color="primary"
          />
        }
        data-value="option1"
        disabled={false}
        key="radio-option1"
        label="label1"
        value="option1"
      />
      <ForwardRef(FormControlLabel)
        className="makeStyles-formControlStyle-4"
        classes={
          Object {
            "label": "makeStyles-radioItemLabel-6",
          }
        }
        control={
          <ForwardRef(Radio)
            classes={
              Object {
                "root": false,
              }
            }
            color="primary"
          />
        }
        data-value="option2"
        disabled={false}
        key="radio-option2"
        label="label2"
        value="option2"
      />
    </ForwardRef(RadioGroup)>
  </div>
</Fragment>
`;
